### PR TITLE
add helm support

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards KPI Monitoring.

### CODE REVIEW

File: helm/github-exporter-chart/templates/configmap.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  namespace: {{ .Values.namespace }}
+data:
+  config.yml: |
+    server:
+      listen:
+        address: 'http://0.0.0.0:8000'
+    service:
+      collector:
+        issue:
+          customlabels: '[ "kind/okr,goal/achieved", "kind/okr,goal/missed", "postmortem,team/batman", "postmortem,team/magic", "postmortem,team/ludacris", "postmortem,team/firecracker" ]'
```
Comment (xh3b4sd) [inline helm/github-exporter-chart/templates/configmap.yaml:14]: Here we can define what special combinations to collect metrics for.

File: helm/github-exporter-chart/templates/rbac.yaml [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+  - apiGroups:
+      - core.giantswarm.io
+    resources:
+      - storageconfigs
+    verbs:
+      - "*"
+  - apiGroups:
+      - provider.giantswarm.io
+    resources:
+      - kvmconfigs
```
Comment (xh3b4sd) [inline helm/github-exporter-chart/templates/rbac.yaml]: I copied that from the `kvm-operator`. How should that look like ideally?

File: helm/github-exporter-chart/templates/service.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: github-exporter
+  namespace: {{ .Values.namespace }}
+  labels:
+    app: github-exporter
+  annotations:
+    giantswarm.io/exporter: "true"
```
Comment (xh3b4sd) [inline helm/github-exporter-chart/templates/service.yaml:8]: Is that right? Should that be a label or annotation?

File: helm/github-exporter-chart/templates/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+          timeoutSeconds: 1
+        readinessProbe:
+          httpGet:
+            path: /healthz
+            port: 8000
+          initialDelaySeconds: 15
+          timeoutSeconds: 1
+        resources:
+          requests:
+            cpu: 250m
```
Comment (JosephSalisbury) [inline helm/github-exporter-chart/templates/deployment.yaml]: This seems kind of high, no?

File: helm/github-exporter-chart/templates/psp.yaml [hunk 5]
```diff
@@ -0,0 +1,20 @@
+apiVersion: extensions/v1beta1
+kind: PodSecurityPolicy
```
Comment (JosephSalisbury) [inline helm/github-exporter-chart/templates/psp.yaml:2]: Do we need a PSP? What calls does the application make to the Kubernetes API?

File: helm/github-exporter-chart/templates/rbac.yaml (hunk 2 above)
Comment (JosephSalisbury) [reply helm/github-exporter-chart/templates/rbac.yaml]: If the github-exporter doesn't make any calls to the Kubernetes API Server, we don't need `ClusterRole` etc.

File: helm/github-exporter-chart/templates/service.yaml (hunk 3 above)
Comment (JosephSalisbury) [reply helm/github-exporter-chart/templates/service.yaml:8]: Label, not annotation

File: helm/github-exporter-chart/templates/service.yaml [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+    prometheus.io/scrape: "true"
```
Comment (JosephSalisbury) [inline helm/github-exporter-chart/templates/service.yaml]: We don't need this annotation

File: helm/github-exporter-chart/templates/psp.yaml (hunk 5 above)
Comment (xh3b4sd) [reply helm/github-exporter-chart/templates/psp.yaml:2]: I think we need this for the configmap and secret. We declare volumes below. Not sure if it works without. Only copied from some operator.

File: helm/github-exporter-chart/templates/deployment.yaml (hunk 4 above)
Comment (xh3b4sd) [reply helm/github-exporter-chart/templates/deployment.yaml]: What is a more appropriate value here?

File: helm/github-exporter-chart/templates/psp.yaml (hunk 5 above)
Comment (JosephSalisbury) [reply helm/github-exporter-chart/templates/psp.yaml:2]: We'd need this if the Pod was requesting the configmap and/or the secret via the API. We don't need it in this case.

File: helm/github-exporter-chart/templates/deployment.yaml (hunk 4 above)
Comment (JosephSalisbury) [reply helm/github-exporter-chart/templates/deployment.yaml]: 50m? 100m? A small exporter using a quarter of a cpu is big.

Comment (xh3b4sd) [reply helm/github-exporter-chart/templates/deployment.yaml]: Going with 50.

File: helm/github-exporter-chart/templates/psp.yaml (hunk 5 above)
Comment (xh3b4sd) [reply helm/github-exporter-chart/templates/psp.yaml:2]: It says this. Doesn't look like API request. I understand this as being necessary in order to mount volumes into the pod. But I am not sure though. 

```
  volumes:
    - 'secret'
    - 'configMap'
```

**Pull request closed without merging**